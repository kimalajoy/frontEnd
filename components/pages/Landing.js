import React from 'react';
import { TouchableOpacity, View, Image, Text } from 'react-native';

import styles from '../../styles/styles';

export default function Landing({ navigation }) {
  return (
    <View style={ styles.container }>
      <Image
        source={require('../../assets/logo.png')}
        style={ styles.logo }/>
      <TouchableOpacity
        style={[styles.landingButtons, styles.borderRadius, styles.boxShadow]}
        onPress={() => navigation.navigate('TabNavigation', { screen: 'Search'})}
        >
        <Text style={ styles.textLight }>
          Search Plants
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        style={[styles.landingButtons, styles.borderRadius, styles.boxShadow]}
        onPress={() => navigation.navigate('TabNavigation', { screen: 'Garden' })}
        >
        <Text style={ styles.textLight }>
          My Garden
        </Text>
      </TouchableOpacity>
    </View>
  );
}
