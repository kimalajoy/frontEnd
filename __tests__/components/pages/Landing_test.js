import Landing from '../../../components/pages/Landing';
import renderer from 'react-test-renderer';
import React from 'react';


describe("Landing Page", () => {
  let landingPage;

  beforeAll(() => {
    landingPage = renderer.create(<Landing />)
  })

  it('Displays the Logo on the Landing Page', () => {
    expect(landingPage.toJSON().children[0].props.source.testUri).toEqual('../../../assets/logo.png');
  })

  it('Displays a button with text "Search Plants"', () => {
    expect(landingPage.toJSON().children[1].children[0].children[0]).toEqual('Search Plants');
  })

  it('Displays a button with text "My Garden"', () => {
    expect(landingPage.toJSON().children[2].children[0].children[0]).toEqual('My Garden');
  })

  it('Landing Page does render', () => {
    expect(landingPage).toMatchSnapshot();
  })
})
