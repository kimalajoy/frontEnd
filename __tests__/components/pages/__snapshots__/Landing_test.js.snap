// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page Landing Page does render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../assets/logo.png",
      }
    }
    style={
      Object {
        "height": 300,
        "width": 300,
      }
    }
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#5DA25B",
        "borderRadius": 10,
        "margin": 10,
        "opacity": 1,
        "padding": 15,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 7,
          "width": 7,
        },
        "shadowOpacity": 0.1,
        "textAlign": "center",
        "width": 150,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 20,
        }
      }
    >
      Search Plants
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#5DA25B",
        "borderRadius": 10,
        "margin": 10,
        "opacity": 1,
        "padding": 15,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 7,
          "width": 7,
        },
        "shadowOpacity": 0.1,
        "textAlign": "center",
        "width": 150,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 20,
        }
      }
    >
      My Garden
    </Text>
  </View>
</View>
`;
